docs: improve IBC guide (#6472)
cosmos/cosmos-sdk: docs/ibc/custom.md

not want the packet processing to revert. Instead, we may want to encode this failure into the
acknowledgement and finish processing the packet. This will ensure the packet cannot be replayed,
and will also allow the sender module to potentially remediate the situation upon receiving the
acknowledgement. An example of this technique is in the `ibc-transfer` module's
[`OnRecvPacket`](https://github.com/cosmos/cosmos-sdk/tree/master/x/ibc-transfer/module.go).
:::

### Acknowledgements

Modules may optionally commit an acknowledgement upon receiving and processing a packet. This
acknowledgement can then be relayed back to the original sender chain, which can take action
depending on the contents of the acknowledgement.

Just as packet data was opaque to IBC, acknowledgements are similarly opaque. Modules must pass and
receive acknowledegments with the IBC modules as byte strings.

Thus, modules must agree on how to encode/decode acknowledgements. The process of creating an
acknowledgement struct along with encoding and decoding it, is very similar to the packet data
example above.

#### Acknowledging Packets

> If the receiving module writes an ackowledgement while processing the
> packet, a relayer can relay back the acknowledgement to the sender module. The sender module can

Reviewer: ```suggestion
After a module writes an acknowledgement while receiving a packet. a relayer can relay back the acknowledgement to the sender module. The sender module can
```
